products/dataproc/api.yaml

```
@@ -0,0 +1,186 @@
+# Copyright 2019 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Dataproc
+display_name: Cloud Dataproc
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://dataproc.googleapis.com/v1/
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://dataproc.googleapis.com/v1beta2/
+scopes:
+  - https://www.googleapis.com/auth/cloud-identity
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Cloud Dataproc API
+    url: https://cloud.google.com/dataproc/docs/reference/rest/
+objects:
+  - !ruby/object:Api::Resource
+    name: 'AutoscalingPolicy'
+    base_url: "projects/{{project}}/locations/{{region}}/autoscalingPolicies"
+    self_link: "projects/{{project}}/locations/{{region}}/autoscalingPolicies/{{policy_id}}"
```

```suggestion
    self_link: "projects/{{project}}/locations/{{region}}/autoscalingPolicies/{{id}}"
```

Since the field is called `id` by the API, we should give it that value here so that api.yaml remains general. I _think_ we do something with overrides to use `policy_id` as the field name in Terraform, but it's possible that's not the case.

---

This makes complete sense to me, however, it doesn't work if I put `id` instead of `policy_id`, when it does a read `id` is empty and it returns a list of the policies.  Am I missing something else that might make it work?

---

Ah, it may just not work. Normally this field would be `{{name}}` which isn't ever really renamed.

Are you able to override the `self_link` in `terraform.yaml`? I *think* that works.